WordPress/Classes/ViewRelated/Gutenberg/Processors/GutenbergGalleryUploadProcessor.swift

```diff
@@ -28,76 +28,67 @@ class GutenbergGalleryUploadProcessor: Processor {
         static let dataLink = "data-link"
     }
 
-    lazy var imgPostMediaUploadProcessor = HTMLProcessor(for: ImageKeys.name, replacer: { (img) in
-        guard let imgClassAttributeValue = img.attributes[ImageKeys.classAttributes]?.value,
-            case let .string(imgClass) = imgClassAttributeValue else {
-                return nil
-        }
+    func processImgPostMediaUpload(_ element: Element) {
+        let imgTags = try? element.select(ImageKeys.name)
+        imgTags?.forEach {imgTag in
+            guard let imgClass = try? imgTag.attr(ImageKeys.classAttributes) else {
+                return
+            }
 
-        let classAttributes = imgClass.components(separatedBy: " ")
+            let classAttributes = imgClass.components(separatedBy: " ")
 
-        guard let imageIDAttribute = classAttributes.filter({ (value) -> Bool in
-            value.hasPrefix(GutenbergImgUploadProcessor.imgClassIDPrefixAttribute)
-        }).first else {
-            return nil
-        }
+            guard let imageIDAttribute = classAttributes.filter({ (value) -> Bool in
+                value.hasPrefix(GutenbergImgUploadProcessor.imgClassIDPrefixAttribute)
+            }).first else {
+                return
+            }
 
-        let imageIDString = String(imageIDAttribute.dropFirst(ImageKeys.classIDPrefix.count))
-        let imgUploadID = Int32(imageIDString)
+            let imageIDString = String(imageIDAttribute.dropFirst(ImageKeys.classIDPrefix.count))
+            let imgUploadID = Int32(imageIDString)
 
-        guard imgUploadID == self.mediaUploadID else {
-            return nil
-        }
+            guard imgUploadID == self.mediaUploadID else {
+                return
+            }
 
-        let newImgClassAttributes = imgClass.replacingOccurrences(of: imageIDAttribute, with: ImageKeys.classIDPrefix + String(self.serverMediaID))
+            let newImgClassAttributes = imgClass.replacingOccurrences(of: imageIDAttribute, with: ImageKeys.classIDPrefix + String(self.serverMediaID))
 
-        var attributes = img.attributes
-        attributes.set(.string(self.remoteURLString), forKey: "src")
-        attributes.set(.string(newImgClassAttributes), forKey: "class")
-        attributes.set(.string("\(self.serverMediaID)"), forKey: ImageKeys.dataID)
-        attributes.set(.string(self.remoteURLString), forKey: ImageKeys.dataFullURL)
-        if attributes.contains(where: { $0.key == ImageKeys.dataLink } ) {
-            attributes.set(.string(self.mediaLink), forKey: ImageKeys.dataLink)
-        }
+            _ = try? imgTag.attr("src", self.remoteURLString)
+            _ = try? imgTag.attr("class", newImgClassAttributes)
+            _ = try? imgTag.attr(ImageKeys.dataID, "\(self.serverMediaID)")
```

Comment:
Optional nit and more of a personal pref, but it may be more common to convert the Int -> String via `String(self.serverMediaID)` if you're not interpolating anything else.

Reply:
I applied this suggestion in https://github.com/wordpress-mobile/WordPress-iOS/pull/22896/commits/7694cb27bc4f662a08faa6745d40c2e67f9577a4.